deps/quickjs/src/quickjs.c

```diff
@@ -16226,10 +16264,30 @@ static JSValue JS_CallInternal(JSContext *caller_ctx, JSValueConst func_obj,
 #include "quickjs-opcode.h"
         [ OP_COUNT ... 255 ] = &&case_default
     };
+#ifndef CONFIG_QUICKJS_DEBUGGER
 #define SWITCH(pc)      goto *dispatch_table[opcode = *pc++];
 #define CASE(op)        case_ ## op
 #define DEFAULT         case_default
 #define BREAK           SWITCH(pc)
+#else    
+    static const void * const debugger_dispatch_table[256] = {
+#define DEF(id, size, n_pop, n_push, f) && case_debugger_OP_ ## id,
+#if SHORT_OPCODES
+#define def(id, size, n_pop, n_push, f)
+#else
+#define def(id, size, n_pop, n_push, f) && case_default,
+#endif
+#include "quickjs-opcode.h"
+        [ OP_COUNT ... 255 ] = &&case_default
+    };
+#define SWITCH(pc)      goto *active_dispatch_table[opcode = *pc++];
+#define CASE(op)        case_debugger_ ## op: js_debugger_check(ctx, pc); case_ ## op
+#define DEFAULT         case_default
+#define BREAK           SWITCH(pc)
+
+    const void * const * active_dispatch_table = caller_ctx->rt->debugger_info.transport_close
+        ? debugger_dispatch_table : dispatch_table;
```

Comment:
Doesn't adding this check plus several pointer indirections introduce pretty hefty overhead to bytecode dispatch? quickjs bytecodes are "fat" but dispatch is often still 10-20% of running time and this change likely makes it worse. Maybe run some before/after benchmarks?

On a side note: I believe SHORT_OPCODES=0 and OPTIMIZE=0 are broken and DIRECT_DISPATCH=0 (emscripten support) probably isn't relevant to txiki so maybe the ifdef-ery here can be simplified somewhat.